mani_skill/envs/tasks/tabletop/pull_cube_tool.py

```diff
@@ -0,0 +1,278 @@
+            tool_xyz[..., 2] = self.height / 2  # place tool on table
+            tool_q = [1, 0, 0, 0]  # no rotation
+
+            tool_pose = Pose.create_from_pq(p=tool_xyz, q=tool_q)
+            self.l_shape_tool.set_pose(tool_pose)
+
+            # Initialize the cube a bit away from the base of the arm
+
+            cube_xyz = torch.zeros((b, 3))
+            cube_xyz[..., 0] = self.arm_reach + torch.rand(b) * (
+                self.handle_length - 0.08
+            )  # Just outside arm's reach
+            cube_xyz[..., 1] = torch.rand(b) * 0.4 - 0.2  # Random y position
+            cube_xyz[..., 2] = self.cube_size / 2  # Place on the table
+
+            cube_q = randomization.random_quaternions(
+                b,
+                lock_x=True,
+                lock_y=True,
+                lock_z=False,
+                bounds=(-np.pi / 6, np.pi / 6),
+            )
+
+            cube_pose = Pose.create_from_pq(p=cube_xyz, q=cube_q)
+            self.cube.set_pose(cube_pose)
+
+    def _get_obs_extra(self, info: Dict):
+        obs = dict(
+            tcp_pose=self.agent.tcp.pose.raw_pose,
+            cube_pose=self.cube.pose.raw_pose,
+            tool_pose=self.l_shape_tool.pose.raw_pose,
+        )
+
+        return obs
+
+    def evaluate(self):
+
+        tcp_pos = self.agent.tcp.pose.p
+        cube_pos = self.cube.pose.p
+
+        cube_in_reach = (
+            torch.linalg.norm(tcp_pos[:, :2] - cube_pos[:, :2], dim=1)
+            < self.goal_radius
+        )
+        cube_picked = self.agent.is_grasping(self.cube)
+
+        return {
+            "success": cube_picked,
+            "cube_in_reach": cube_in_reach,
+        }
+
+    def compute_dense_reward(self, obs: Any, action: torch.Tensor, info: Dict):
+        
+        tcp_pos = self.agent.tcp.pose.p
+        cube_pos = self.cube.pose.p
+        tool_pos = self.l_shape_tool.pose.p
+
+        # Reward for reaching the tool
+        tcp_to_tool_dist = torch.linalg.norm(tcp_pos - tool_pos, dim=1)
+        reaching_tool_reward = 1 - torch.tanh(5.0 * tcp_to_tool_dist)
+
+        # Reward for moving the tool towards the cube
+        tool_to_cube_dist = torch.linalg.norm(tool_pos - cube_pos, dim=1)
+        tool_cube_reward = 1 - torch.tanh(5.0 * tool_to_cube_dist)
+
+        # Reward for bringing the cube closer to the robot
+        tcp_to_cube_dist = torch.linalg.norm(tcp_pos - cube_pos, dim=1)
+        cube_close_reward = 1 - torch.tanh(5.0 * tcp_to_cube_dist)
+
+        # Success reward
+        success_reward = info["success"].float() * 10
+
+        # Combine rewards
+        reward = (
+            reaching_tool_reward + tool_cube_reward + cube_close_reward + success_reward
+        )
+
+        return reward
+
+    def compute_normalized_dense_reward(
+        self, obs: Any, action: torch.Tensor, info: Dict
+    ):
+        max_reward = 13.0  # 10 + 1 + 1 + 1
+        return self.compute_dense_reward(obs=obs, action=action, info=info) / max_reward
```

Comment:
have you tested the reward function with PPO? If so can you add an example script to examples/baselines/ppo/examples.sh that works?